Analysis/Tasks/pidTOF_tiny.cxx

```diff
@@ -0,0 +1,113 @@
+    {"pid-pikapr", VariantType::Int, 1, {"Produce PID information for the Pion, Kaon, Proton mass hypothesis"}},
+    {"pid-nuclei", VariantType::Int, 1, {"Produce PID information for the Deuteron, Triton, Alpha mass hypothesis"}}};
+  std::swap(workflowOptions, options);
+}
+
+#include "Framework/runDataProcessing.h"
+
+template <o2::track::PID::ID pid_type, typename table>
+struct pidTOFTaskPerTiny {
+  using Trks = soa::Join<aod::Tracks, aod::TracksExtra>;
+  using Coll = aod::Collisions;
+  Produces<table> tofpid;
+  DetectorResponse resp;
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+  Configurable<std::string> paramfile{"param-file", "", "Path to the parametrization object, if emtpy the parametrization is not taken from file"};
+  Configurable<std::string> sigmaname{"param-sigma", "TOFReso", "Name of the parametrization for the expected sigma, used in both file and CCDB mode"};
+  Configurable<std::string> url{"ccdb-url", "http://ccdb-test.cern.ch:8080", "url of the ccdb repository"};
+  Configurable<long> timestamp{"ccdb-timestamp", -1, "timestamp of the object"};
+
+  void init(o2::framework::InitContext&)
+  {
+    ccdb->setURL(url.value);
+    ccdb->setTimestamp(timestamp.value);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    // Not later than now objects
+    ccdb->setCreatedNotAfter(std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+    //
+    const std::vector<float> p = {0.008, 0.008, 0.002, 40.0};
+    resp.SetParameters(DetectorResponse::kSigma, p);
+    const std::string fname = paramfile.value;
+    if (!fname.empty()) { // Loading the parametrization from file
+      resp.LoadParamFromFile(fname.data(), sigmaname.value, DetectorResponse::kSigma);
+    } else { // Loading it from CCDB
+      const std::string path = "Analysis/PID/TOF";
+      resp.LoadParam(DetectorResponse::kSigma, ccdb->getForTimeStamp<Parametrization>(path + "/" + sigmaname.value, timestamp.value));
+    }
+  }
+
+  void process(Coll const& collisions, Trks const& tracks)
+  {
+    constexpr tof::ExpTimes<Coll::iterator, Trks::iterator, pid_type> resp_PID = tof::ExpTimes<Coll::iterator, Trks::iterator, pid_type>();
+
+    tofpid.reserve(tracks.size());
+    for (auto const& trk : tracks) {
+      if (resp_PID.GetExpectedSigma(resp, trk.collision(), trk) <= o2::aod::pidtof_tiny::bin_min) {
+        tofpid(resp_PID.GetExpectedSigma(resp, trk.collision(), trk),
+               0);
+      } else if (resp_PID.GetExpectedSigma(resp, trk.collision(), trk) >= o2::aod::pidtof_tiny::bin_max) {
+        tofpid(resp_PID.GetExpectedSigma(resp, trk.collision(), trk),
+               o2::aod::pidtof_tiny::nbins);
+      } else {
+        tofpid(resp_PID.GetExpectedSigma(resp, trk.collision(), trk),
+               (resp_PID.GetSeparation(resp, trk.collision(), trk) - o2::aod::pidtof_tiny::bin_min) / (o2::aod::pidtof_tiny::bin_width)*o2::aod::pidtof_tiny::nbins);
+      }
```

Comment:
In this way the resp_PID.GetExpectedSigma is computed up to three times. Better put it in a variable before. Does this conflict with an earlier comment? ;-)

Reply:
I don't know what I was thinking, I probably got confused and thought it was coming from a table ;) However, it is fixed now